crates/ra_assists/src/handlers/replace_unwrap_with_match.rs

```diff
@@ -46,37 +46,44 @@ pub(crate) fn replace_unwrap_with_match(acc: &mut Assists, ctx: &AssistContext)
     let ty = ctx.sema.type_of_expr(&caller)?;
     let happy_variant = TryEnum::from_ty(&ctx.sema, &ty)?.happy_case();
     let target = method_call.syntax().text_range();
-    acc.add(AssistId("replace_unwrap_with_match"), "Replace unwrap with match", target, |builder| {
-        let ok_path = make::path_unqualified(make::path_segment(make::name_ref(happy_variant)));
-        let it = make::bind_pat(make::name("a")).into();
-        let ok_tuple = make::tuple_struct_pat(ok_path, iter::once(it)).into();
+    acc.add(
+        AssistId("replace_unwrap_with_match"),
+        AssistKind::RefactorRewrite,
```

Comment:
Hm, let's introduce two kinds for:

* assist that rewrite code to equivalent (apply de-morgan, replace if-let with match, etc)
* assists that may change semantics of code

Not sure what are the best names for two things...

Reply:
According to the lsp spec:

> The set of kinds is open and client needs to announce the kinds it supports to the server during initialization.

This means that at least if you introduce new kinds, you would have to support a fallback when the client doesn't say it supports them.

Reply:
> This means that at least if you introduce new kinds, you would have to support a fallback when the client doesn't say it supports them.

It's the client job to handle unknown values:

> * The code action kind values the client supports. When this
>  * property exists the client also guarantees that it will
>  * handle values outside its set gracefully and falls back
>   * to a default value when unknown.